diff --git a/src/generator.c b/src/generator.c
index 22e2e080..ea36cc0d 100644
--- a/src/generator.c
+++ b/src/generator.c
@@ -1947,7 +1947,7 @@ static void *proxy_send(void *arg)
 		}
 
 		mutex_lock(&gdata->notify_lock);
-		HASH_FIND_INT(gdata->notify_instances, &id, ni);
+		HASH_FIND_I64(gdata->notify_instances, &id, ni);
 		if (ni)
 			jobid = json_copy(ni->jobid);
 		mutex_unlock(&gdata->notify_lock);
